services/motion/builtin/move_request.go

```diff
@@ -267,6 +293,9 @@ func (ms *builtIn) newMoveOnGlobeRequest(
 	if straightlineDistance > maxTravelDistanceMM {
 		return nil, fmt.Errorf("cannot move more than %d kilometers", int(maxTravelDistanceMM*1e-6))
 	}
+	// NOTE: I don't understand what a referenceframe.Limit & that the mins & maxes describe.
+	// NOTE: Why why are we setting these limits to the distance we plan on traveling in mm * +-3 in the first 2 limits
+	// NOTE: Why are we setting the last one to +-2PI
```

Comment:
These are legacy values that are persisted only to support `diffDriveKinematics.go`, and refer to the inputs used there; specifically moving in terms of `x, y, theta`.

As we plan for that type, we are saying "don't try to create a path that goes further from the starting point than 3x the distance to the goal", so that RRT has a "reasonable" area in which to sample.

Also known as "don't go from Manhattan to Brooklyn via Boston"

Reply:
That makes sense. Does this imply that if there are so many obstacles that the robot would take a path more than 3x the straight line distance from source to dest, planning is expected to fail currently?

Reply:
Only if someone implements their own motion service that uses `diffDriveKinematics`; that code is mostly unused right now aside from tests and Jeremy's new `Explore` service.

PTG kinematics do not have the abovementioned limitation.